flixel/system/render/hardware/gl/HardwareRenderer.hx

```diff
@@ -0,0 +1,322 @@
+package flixel.system.render.hardware.gl;
+
+import flixel.graphics.FlxGraphic;
+import flixel.graphics.shaders.FlxColorShader;
+import flixel.graphics.shaders.FlxTexturedShader;
+import flixel.system.FlxAssets.FlxShader;
+import flixel.util.FlxDestroyUtil;
+import flixel.util.FlxDestroyUtil.IFlxDestroyable;
+import openfl.geom.ColorTransform;
+
+#if FLX_RENDER_GL
+import lime.graphics.GLRenderContext;
+import lime.utils.Float32Array;
+import openfl._internal.renderer.RenderSession;
+import openfl._internal.renderer.opengl.GLRenderer;
+#end
+
+import openfl.display.BlendMode;
+import openfl.display.DisplayObject;
+import openfl.geom.Matrix;
+import openfl.geom.Rectangle;
+
+// TODO: try to add general vertex and index arrays to minimize data upload operations (gl.bufferData() calls). Like it's done in GL implementation of Tilemap renderer...
+// TODO: multitexture batching...
+// TODO: sprite materials with multiple textures...
+// TODO: support for colorOffsets???
+
+/**
+ * ...
+ * @author Yanrishatum
```

Comment:
it's Pavel Alexandrov aka Yanrishatum https://github.com/Yanrishatum 
i took big part from his fork of HaxePunk and he didn't mind that i used his work as a basis
Will add links in the next commit